### PR TITLE
Em rg/search users

### PR DESCRIPTION
**Pull request by greatnessmensah**

### CODE REVIEW

**Review by RansfordGenesis — APPROVED** [review]
Great

File: api/routes/profile_page.py [hunk 1]
```diff
@@ -45,36 +50,38 @@ async def update_profile(userDetails: ProfileUpdate, current_user: User = Depend
...
-        pages=pages,
-        links=links,
-    )
+@profile_route.get("/", response_model=Page[ProfileResponse])
+def get_all_profile(skill: str = Query(None, title="Skill", description="The skill to filter users"), 
+                    stack: str = Query(None, title="Stack", description="The stack to filter users"),
+                    active: Optional[bool] = None, p: Optional[str] = None,
+                    db: Session = Depends(get_db)):
 
+    query = db.query(User)
```
Comment (TMCreme) [inline api/routes/profile_page.py:59]: It's a good idea to have a base query here. But I don't think the code has made provision for multiple parameters. What if I want to search for active users with Backend stack and Python skill? I think the base should be overridden in each condition. Or how ever you think it should be done.

File: api/routes/profile_page.py [hunk 2]
```diff
@@ -45,36 +50,38 @@ async def update_profile(userDetails: ProfileUpdate, current_user: User = Depend
...
+    
+    if skill:
+        users_query = query.join(users_skills.UserSkill).join(Skill).filter(Skill.name == skill.capitalize())
+        return paginate(users_query.order_by(desc(User.created_at)))
+    
+    if stack:
+        users_query = query.filter(User.stack.has(name=stack.capitalize()))
+        return paginate(users_query.order_by(desc(User.created_at)))
+    
+    if active is not None:
```
Comment (TMCreme) [inline api/routes/profile_page.py:69]: In this if-else block, you entirely ignored the base query (query = db.query(User)). I'm not sure if it's an oversight or intentional. Can you clarify? If it's not intentional, I think using the base query here will be better.

Comment (greatnessmensah) [reply api/routes/profile_page.py:69]: I guess it is an oversight, I will fix that.

**Review by TMCreme — CHANGES_REQUESTED** [review]
Excellent work. I have left some comments. Kindly check. Thank you.

**Pull request merged**